service/controller/app/v1/resource/secret/desired.go

```diff
@@ -55,6 +56,21 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	// We get the user level values if configured and merge them.
+	if userSecretName != "" {
+		userData, err := r.getUserSecretDataForApp(ctx, cr)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		// Secrets are merged again and in case of intersecting values the user
+		// level secrets are preferred.
+		mergedData, err = values.MergeSecretData(mergedData, userData)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
```

Comment:
We do a 2nd merge if a user secret has been configured.

The 2 step process is because most app CRs won't have user config. It is only for managed apps where we or the customer need to override values per tenant cluster.